actor/src/main/scala-2.12/org/apache/pekko/util/ByteIterator.scala

```diff
@@ -484,10 +484,10 @@ abstract class ByteIterator extends BufferedIterator[Byte] {
   }
 
   def indexOf(elem: Byte): Int = indexOf(elem, 0)
-  def indexOf(elem: Byte, from: Int): Int = indexWhere(_ == elem, from)
+  def indexOf(elem: Byte, from: Int): Int = indexWhere(elem == _, from)
```

Comment:
Good question. I was going to point to the stdlib, but apparently that's also inconsistent:
- Iterator uses `_ == elem`: https://github.com/scala/scala/blob/v2.13.10/src/library/scala/collection/Iterator.scala#L449-L461
- `Seq` uses `elem == _`: https://github.com/scala/scala/blob/v2.13.10/src/library/scala/collection/Seq.scala#L365

Although I believe that `elem == _` makes more sense. Otherwise it feels a bit useless to define `indexOf[B >: A]` 🤷

Reply:
That's ture, I think we should open up an issue in scala/bug before we merge this, because this is kind of behavior change in anyway.

Reply:
So how about we split this PR into two ？

Reply:
I think I can remove this change. From the Java documentation: https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#equals-java.lang.Object-

>  The equals method implements an equivalence relation on non-null object references: 
> [...]
> - It is symmetric: for any non-null reference values x and y, x.equals(y) should return true if and only if y.equals(x) returns true.

So I think it's OK to assume that the order doesn't matter and it's OK to have undefined behavior for wrong implementations of equals.

Reply:
Just force pushed this branch without the problematic commit.